proto/process/agent.proto

```diff
@@ -97,12 +96,35 @@ message CollectorConnections {
 
 	map<string, RuntimeCompilationTelemetry> compilationTelemetryByAsset = 21;
 
+	repeated Route routes = 31;
+
+	repeated RouteMetadata routeMetadata = 34;
+
+	/////////////////
+	// V1 encodings
+
+	// encoded dns is a map of (string) ip-> list of domains (DNSEntry)
+	bytes encodedDNS = 14;
+
 	// All queried DNS domains for which we have collected stats
 	repeated string domains = 30;
 
-	repeated Route routes = 31;
+	//////////////////
+	// V2 encodings
+	// all domain strings are now indexes into a single array of strings
+	// hostname query information.  Encoded via V2DNSEncoder.EncodeDomainDatabase
+	// is a single buffer; varint number of strings, followed by each string.
+	bytes encodedDomainDatabase = 35;
+
+	// new representation of encoded DNS.  This is a map of the string representation
+	// of the IP --> DNSDatabaseEntry message. (was encodedDNS).  Encoded
+	// via V2DNSEncoder.EncodeMapped.  Full buffer description provided there.
+	bytes encodedDnsLookups = 36;
+
+	// All queried DNS domains for which we have collected stats, integers are indexes
+	// into the encodedDomainDatabase (was domains)
+	repeated int32 indexedDomains = 37;
```

Comment:
@mbakht does the backend actually need this field ?  We should be able to loop through all connections and get the domains from there.

Reply:
No we don’t. @derekwbrown and I had a chat on Slack yesterday about it and I explained why we don’t need it.